Added support to also generate paths of YANG containers

fgvdeput opened the pull request:
I've added a command flag '--with-containers' to optionally also generate paths for the intermediate container nodes, instead of only generating paths for the leaf nodes.

hellt commented:
Thanks @fgvdeput 
I believe this is in the context of parsing containers descriptions that we discussed yesterday that can greatly improve the dataset our AI engine works on?

hellt commented on config/config.go:
```diff
@@ -197,13 +197,14 @@ type LocalFlags struct {
+	GeneratePathPathType       string `mapstructure:"generate-path-path-type,omitempty" json:"generate-path-path-type,omitempty" yaml:"generate-path-path-type,omitempty"`
+	GeneratePathState          bool   `mapstructure:"generate-path-state,omitempty" json:"generate-path-state,omitempty" yaml:"generate-path-state,omitempty"`
+	GeneratePathConfig         bool   `mapstructure:"generate-path-config,omitempty" json:"generate-path-config,omitempty" yaml:"generate-path-config,omitempty"`
+	GeneratePathWithContainers bool   `mapstructure:"generate-path-with-containers,omitempty" json:"generate-path-with-containers,omitempty" yaml:"generate-path-with-containers,omitempty"`
```
could you please also add a doc entry for this new option https://github.com/karimra/gnmic/blob/main/docs/cmd/path.md ?

fgvdeput replied:
Indeed, this seems to enable this capability (cfr also the yang-browser repo). 

I've added a doc entry for the new option.

fgvdeput commented:
I've fixed the issue with YANG cases and choices. 

Moreover, because not only container paths are returned, but also list paths, I've renamed the command flag to '--with-non-leaves', which is more generic. 

Finally, because 'container' and 'list' are not actual leaf data types, but rather schema element types, I've added brackets around them. Other suggestions are welcome:

```/acl    (type=[container])  
/acl/capture-filter     (type=[container])
/acl/capture-filter/ipv4-filter (type=[container])
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]    (type=[list])
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]/action     (type=[container])
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]/action/accept      (type=[container])
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]/action/copy        (type=[container])
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]/description        (type=description)
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]/match      (type=[container])
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]/match/destination-ip       (type=[container])
/acl/capture-filter/ipv4-filter/entry[sequence-id=*]/match/destination-ip/address       (type=ipv4-address)
```

karimra commented:
@fgvdeput thanks for the contribution and sorry for the delay in merging it.
@hellt thanks for the review
